### PR TITLE
Use upstream version of keytar

### PR DESCRIPTION
**Pull request by RomanNikitenko**
### What does this PR do?
- as far as I understand
`node-addon-api` and `node-addon-api-latest`are transitive dependencies. They were added to `devDependencies`section. I believe the dependencies should be removed from the `devDependencies` section.
- some time ago there were some problems in `theia` related to `keytar` version. They had to downgrade the dependency to 7.2.0 version. I think it makes sense to use the same version in `che-theia`.

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)

- [ ] [The Eclipse Contributor Agreement is valid](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-eclipse-contributor-agreement-is-valid)
- [ ] [Code produced is complete](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-produced-is-complete)
- [ ] [Code builds without errors](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-builds-without-errors)
- [ ] [Tests are covering the bugfix](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#tests-are-covering-the-bugfix)
- [ ] [The repository devfile is up to date and works](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-repository-devfile-is-up-to-date-and-works)
- [ ] [Sections `What issues does this PR fix or reference` and `How to test this PR` completed](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#sections-what-issues-does-this-pr-fix-or-reference-and-how-to-test-this-pr-completed)
- [ ] [Relevant user documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [Relevant contributing documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [CI/CD changes implemented, documented and communicated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#cicd-changes-implemented-documented-and-communicated)
- [ ] Optional Companion PR for updating the HappyPath tests is approved and ready to be merged

### Reviewers

Reviewers, please comment how you tested the PR when approving it.

### Happy Path Channel

HAPPY_PATH_CHANNEL=stable

### CODE REVIEW

File: yarn.lock [hunk 1]
```diff
@@ -11702,15 +11702,20 @@ node-abi@^2.21.0, node-abi@^2.7.0:
   dependencies:
     semver "^5.4.1"
 
+node-addon-api@*:
```
Comment (nickboldt) [inline yarn.lock:11705]: need to find out of the resulting crw theia yarn tarball will include BOTH node-addon-api 4.3.0 and 3.2.1, and if that will break anything (keytar, lerna).

**Comment by nickboldt** [comment]
Seems legit, with one question (re: two different versions of node-addon-api).

**Review by nickboldt — APPROVED** [review]

File: yarn.lock (hunk 1 above)
Comment (RomanNikitenko) [reply yarn.lock:11705]: the similar state in upstream
- node-addon-api@*: https://github.com/eclipse-theia/theia/blob/f81ef73a5379c938f57aad5ba0a109964169b4a4/yarn.lock#L7560
- node-addon-api@^3.0.0 https://github.com/eclipse-theia/theia/blob/f81ef73a5379c938f57aad5ba0a109964169b4a4/yarn.lock#L7565

**Comment by RomanNikitenko** [comment]
let's check the changes for `7.42.x` first
https://github.com/eclipse-che/che-theia/pull/1309

**Comment by nickboldt** [comment]
https://github.com/eclipse-che/che-theia/pull/1309/files has passed, so I'm going to merge this as is. 

NOTE: there are some small differences between 7.42.x yarnl.lock and this one, which hopefully don't matter. 

Left: main w/ this PR applied, right: 7.42.x branch:

image

**Pull request merged**